Review проекта "Оно тебе надо"

Pololelka opened the pull request.

Pololelka committed:
Первый коммит проекта

Pololelka committed:
Добавление тега main

seginel commented on index.html:
```diff
@@ -1,10 +1,123 @@
-  </head>
-  <body></body>
-</html>
+</head>
```
Отлично. head правильно заполнен

seginel commented on index.html:
```diff
@@ -1,10 +1,123 @@
+                        <article class="card card_type_picture">
+                            <h3 class="card__title">Картина, которую повторит даже 5-летний сын маминой подруги</h3>
+                            <p class="card__text">Ну действительно, там линия, тут кружочек, а здесь и вовсе что-то похожее на инфузорию-туфельку. Никаких полей, лесов, котиков<span
+                                    class="letter-spacing-extended-0-1">...</span> Сколько бы вы на такое
```
Надо исправить.
В именах классов не стоит использовать числа (1, 2, 3 ...), числительные (first, second, ...) или строковые представления чисел (one, two,...). Такая необходимость возникает крайне редко и обычно подобные имена явно требуются в задании на проект. Стоит помнить, что элементы по разным причинам могут менять свой порядок и элемент с классом `1` или `first` может, например, оказаться на втором месте. В именах классов прежде всего нужно отражать смысл, а не их порядковый номер в блоке.

Но по большому счёту в этом span и в этом классе нет необходимости. Разные letter-spacing внутри параграфа - это скорее ошибка дизайна, чем задумка дизайнера

seginel commented on index.html:
```diff
@@ -1,10 +1,123 @@
+            </ul>
+        </nav>
+        <ul class="footer__social-list">
+            <li><a><img class="footer__social-icon" src="images/yt.svg" alt=""></a></li>
```
Можно лучше: 
Атрибут alt хорошо бы заполнить. Он является подсказкой для пользователя, который не может увидеть изображение https://habr.com/ru/post/341810/

seginel commented on index.html:
```diff
@@ -1,10 +1,123 @@
+    </header>
+    <main>
+        <section class="cover">
+            <h1 class="cover__title">
```
Отлично
Это основной заголовок страницы. Для него верно выбран тег первого уровня. Помните, что заголовок первого уровня может быть только один на странице.

seginel commented on index.html:
```diff
@@ -1,10 +1,123 @@
+                <li class="lots__card-list-item">
+                    <a class="card-link" href="#type_film">
+                        <article class="card card_type_film">
+                            <h3 class="card__title">Фильм режиссёра, который бросил киношколу</h3>
```
Отлично 
Соблюдена многоуровневость заголовков. Заголовки реализованы в соответствии с иерархией страницы.

seginel commented on index.html:
```diff
@@ -1,10 +1,123 @@
+            <br>наб. Принсенграхт 263-<br>265, Амстердам
+        </address>
+    </header>
+    <main>
```
Отлично - основной контент обёрнут в тег `main`

seginel commented on index.html:
```diff
@@ -1,10 +1,123 @@
+        <address class="address">
+            <a href="tel:+99995555555">+9 999 555 5555</a><br>
+            <a href="mailto:info@sobaka.ge">info@sobaka.ge</a>
+            <br>наб. Принсенграхт 263-<br>265, Амстердам
```
Это текстовый фрагмент, его нужно обернуть в тег параграфа `<p>`.

seginel commented on index.html:
```diff
@@ -1,10 +1,123 @@
+                <li><a class="header__menu-link" href="#lots">посмотреть лоты</a></li>
+                <li><a class="header__menu-link" href="#about">об аукционе</a></li>
+            </ul>
+            <a class="header__logo__link"><img class="header__logo-image" src="images/logo-black.svg" alt="Логотип"></a>
```
Надо исправить:
Логотип не является навигацией, его не нужно помещать внутрь тега nav

seginel commented on index.html:
```diff
@@ -1,10 +1,123 @@
+            </ul>
+            <a class="header__logo__link"><img class="header__logo-image" src="images/logo-black.svg" alt="Логотип"></a>
+        </nav>
+        <address class="address">
```
Отлично 
Блок с контактной информацией обёрнут семантическим тегом `<address>`

seginel commented on index.html:
```diff
@@ -1,10 +1,123 @@
+            </h1>
+            <div class="cover__description">
+                <p class="cover__description-text">Аукцион вещей, в которые никто не верил</p>
+                <button class="bet-button">Сделать ставку</button>
```
Можно лучше 
У кнопки в проекте хорошо бы указывать атрибут `type` с нужным значением. Подробнее: http://htmlbook.ru/html/button/type. Если данный атрибут не указать, то будет использован тип по-умолчанию `submit`, однако данный тип нужен для кнопок, которые отправляют данные в формах <form>. То есть для кнопок, которые отправляют форму, атрибут можно не указывать, для всех остальных кнопок атрибут стоит указывать. Для обычных кнопок значение атрибута должно быть `button`

seginel commented on index.html:
```diff
@@ -1,10 +1,123 @@
+        </nav>
+        <ul class="footer__social-list">
+            <li><a><img class="footer__social-icon" src="images/yt.svg" alt=""></a></li>
+            <li><a><img class="footer__social-icon" src="images/vk.svg" alt=""></a></li>
```
Можно лучше: 
Ссылкам на внешние сайты можно добавить атрибут target="_blank", чтобы они открывались в новой вкладке. После того как пользователь прочитает интересующую его информацию на внешнем сайте ему будет легко вернуться на ваш сайт - нужно будет просто закрыть вкладку с внешним сайтом.

Здесь стоило бы добавить атрибут `href` у ссылок. В брифе значения для ссылок не указаны, поэтому в качестве заглушки в атрибуте `href` можно добавить значение `#0`

seginel commented on index.html:
```diff
@@ -1,10 +1,123 @@
+        <address class="address">
+            <a href="tel:+99995555555">+9 999 555 5555</a><br>
+            <a href="mailto:info@sobaka.ge">info@sobaka.ge</a>
+            <br>наб. Принсенграхт 263-<br>265, Амстердам
```
Надо исправить
Это текстовый фрагмент, его нужно обернуть в тег параграфа `<p>`.

seginel commented on styles/style.css:
```diff
@@ -1 +1,313 @@
-/* место для вашего кода */
+html {
+    background-color: #4c4c4c;
```
Надо исправить: такого требования нет в задании и в брифе - оставьте здесь цвет по-умолчанию, какой указан в макете, чисто белый

seginel commented on styles/style.css:
```diff
@@ -1 +1,313 @@
-/* место для вашего кода */
+html {
```
Надо исправить 
стиль для html- являются общими - их стоит прописать в файле global.css

seginel commented on styles/style.css:
```diff
@@ -1 +1,313 @@
+
+.cover {
+    background-image: url(../images/cover.jpg);
+    background-size: cover;
```
Надо исправить 
Для корректного отображения по макету фоновому изображению в блоке нужен стиль `background-position: center;` + `background-size: cover;`

seginel commented on styles/style.css:
```diff
@@ -1 +1,313 @@
+    text-transform: uppercase;
+}
+
+.bet-button {
```
Можно лучше: 
Это элемент взаимодействия с пользователем. Ему хорошо бы добавить свойство `cursor: pointer`, чтобы при наведении курсор менялся, оповещая пользователя, что это интерактивный элемент и на него можно нажать

Надо исправить: Шрифт у кнопки не по макету. Кнопке применяется браузерный шрифт по-умолчанию Arial https://disk.yandex.ru/i/wmA-RcVM_1jiBQ. Несмотря на заданный `font-family` у `body` для кнопок нужный шрифт не наследуется. Для кнопки нужно явно прописать шрифты в свойства font-family (не забыв указать тип засечек)

seginel commented on styles/style.css:
```diff
@@ -1 +1,313 @@
+}
+
+.card__text {
+    position: absolute;
```
Надо исправить
Aбсолютное позиционирование должно использоваться только там, где нельзя применить статичное или относительное позиционирование

seginel commented on styles/style.css:
```diff
@@ -1 +1,313 @@
+    background-size: cover;
+    background-repeat: no-repeat;
+    background-position: top;
+    max-height: 757px;
```
Надо исправить
Здесь стоит убрать жёсткое ограничение высоты, высота должна зависеть от количества контента внутри контейнера. Контента может быть и больше. Здесь подойдёт более мягкое свойство `min-height`, которое не позволяет ужиматься блоку меньше определенной высоты даже в том случае если в нем будет мало контента. В отличие от других мест, где высоту стоит вообще удалять, здесь высоту нужно заменить на `min-height`, так как в данной секции есть фоновая картинка, которая визуально обозначает границы секции и не должна ужиматься, чтобы не нарушать внешний вид страницы по макету.

seginel commented on styles/style.css:
```diff
@@ -1 +1,313 @@
+
+.header {
+    width: 100%;
+    max-height: 310px;
```
Нужно убрать ограничение высоты, высота должна зависеть от количества контента внутри контейнера. При добавлении каких-то новых элементов высота может стать больше.

seginel reviewed (changes requested):
Здравствуйте, У вас получилась хорошая работа, но в ней есть замечания - здесь понадобятся доработки.

### Что понравилось

* Аккуратное форматирование кода
* Используются сематические теги
* Размечена вся структура
* Страница визуально похожа на макет
* Страница валидна
* Правильно оформлен head

### Что нужно поправить

* Все комментарии "Надо исправить" в коде.
* При правках и написании кода придерживайтесь [общих принципов написания кода в Яндекс.Практикуме](https://code.s3.yandex.net/web-developer/verstka/%D0%9A%D0%BE%D0%B4%D1%81%D1%82%D0%B0%D0%B9%D0%BB__%D0%BE%D0%B1%D1%89%D0%B8%D0%B5_%D0%BF%D1%80%D0%B8%D0%BD%D1%86%D0%B8%D0%BF%D1%8B.pdf){target="_blank"}
* После внесения правок проверьте код в валидаторе https://validator.w3.org/nu/#textarea, там не должно быть критических ошибок типа ERROR. При наличии ошибок - их нужно проанализировать и исправить (если эта ошибка не противоречит другим пунктам в чеклиста к проекту).
* Стоит напомнить, что все замечания не придумываются ревьюером на ходу. Они формируются на основе чеклиста и брифа к заданию. Если соблюсти все требования чеклиста и рекомендаций в брифе, то работа будет сразу же принята. Поэтому перед отправкой на следующую итерацию рекомендую самостоятельно пройтись по брифу и чеклисту и при необходимости поправить все проблемные места. Это позволит быстрее получить зачёт по работе.

### Что можно лучше

* Все комментарии "Можно лучше" в коде.

Комментарии "можно лучше" носят рекомендательный характер, их выполнение остается на Ваше усмотрение. Комментарии "надо исправить" являются обязательными, при выполнении всех "надо исправить" работа будет зачтена. Удачи

Pololelka committed:
Исправления от ревьюера

Исправлены замечания, в том числе изменена верстка шапки, вместо флекс использовала гриды, в верстке карточки для текста карточке убрала абсолютное позиционирование и применила флекс

seginel commented on index.html:
```diff
@@ -1,10 +1,124 @@
+            </ul>
+        </nav>
+        <ul class="footer__social-list">
+            <li><a href="#" target="_blank"><img class="footer__social-icon" src="images/yt.svg" alt="Логотип ютуб"></a></li>
```
Отлично - alt у изображений заполнены

seginel commented on index.html:
```diff
@@ -1,10 +1,124 @@
+            </h1>
+            <div class="cover__description">
+                <p class="cover__description-text">Аукцион вещей, в которые никто не верил</p>
+                <button class="bet-button" type="button">Сделать ставку</button>
```
Отлично - у кнопки указан верный тип

seginel reviewed (approved):
Спасибо за правки. Вы внесли исправления по всем замечаниям и хочу поздравить - Вы отлично справились с выполнением проектной работы. На прошлой итерации не обратил внимание на ссылку под лотами: https://disk.yandex.ru/i/m49IdhOgS8RHVg по макету у неё не должно быть двойного подчёркивания. Это стоит поправить. В остальном по странице у меня больше нет замечаний: страница соответствует макету и требованиям чеклиста.
Работу принимаю. Удачи в дальнейшем обучении.

Pololelka committed:
Исправлениее от ревьюера 

убран text-decoration у ссылки
